lib/services/api/session_manager.ts

```diff
@@ -122,40 +122,42 @@ export class SNSessionManager extends PureService {
       {
         onCancel: () => {
           this.isSessionRenewChallengePresented = false;
+        },
+        onNonvalidatedSubmit: async (challengeResponse) => {
```

Comment:
The function itself is not async, it's just that we use `await` inside so I marked it as async.

Reply:
What I mean is that this function does not complete in one JS cycle (because it awaits for something), so users could submit new values while this function was still awaiting a result.